components/analytics/components/RecentSearches.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import get from 'lodash/get';
+import { withRouter } from 'react-router-dom';
+import Searches from './Searches';
+import Filter from './Filter';
+import { exportCSVFile, recentSearchesFull, applyFilterParams } from '../utils';
+import { getAppRecentSearches } from '../../../modules/actions/analytics';
+import Loader from '../../shared/Loader/Spinner';
+import { setFilterValue } from '../../../modules/actions';
+import { withErrorToaster } from '../../shared/ErrorToaster/ErrorToaster';
+
+const headers = {
+	key: 'Search Terms',
+	count: 'Total Queries',
+};
+class RecentSearches extends React.Component {
+	componentDidMount() {
+		const { filterId, filters, selectFilterValue } = this.props;
+		applyFilterParams({
+			filters,
+			filterId,
+			callback: this.fetchRecentSearches,
+			applyFilter: selectFilterValue,
+		});
+	}
+
+	componentDidUpdate(prevProps) {
+		const { filters } = this.props;
```

Comment:
@anik-ghosh-au7 Error handling is missing. https://github.com/appbaseio/batteries/blob/b6259088c1f8c8d3de9456975c5a43f06cc40434/components/analytics/components/RequestDistribution/index.js#L57